### PR TITLE
[SPARK-29113][DOC] Fix some annotation errors and remove meaningless annotations in project.

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

In this PR, I fix some annotation errors and remove meaningless annotations in project. 
### Why are the changes needed?
There are some annotation errors and meaningless annotations in project.
### Does this PR introduce any user-facing change?
No.
### How was this patch tested?
Verified manually.

### CODE REVIEW

**Comment by wangyum** [comment]
ok to test

**Review by wangyum — COMMENTED** [review]
Cloud you find more annotation errors?

**Comment by turboFei** [comment]
I'am reading the code of history server part recently.
And In this PR, I have fix the annotation errors of history server part.
I will try my best to found if there are more annotation errors in whole project.

**Comment by HyukjinKwon** [comment]
Yea, let's at least try to find some more instances while we're here. The changes are fine.

**Comment by srowen** [comment]
Yeah, agree we can fix several more of these. A quick inspection says look at ExplainUtils too.

I'm also in favor of removing empty docs like `@param foo` with no description or `@throws foo` with no detail, as they don't actually do anything. But there are a lot of those. So maybe just the one additional class here.

File: sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/auth/HiveAuthFactory.java [hunk 1]
```diff
@@ -198,9 +198,6 @@ public TTransportFactory getAuthTransFactory() throws LoginException {
 
   /**
    * Returns the thrift processor factory for HiveServer2 running in binary mode
-   * @param service
```
Comment (srowen) [inline sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/auth/HiveAuthFactory.java]: Ah, I forgot to note that we probably don't want to modify this Hive code as it's copied from Hive, and it's easier not to change it. Could you just revert all the changes to these dirs?

File: core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java [hunk 2]
```diff
@@ -276,8 +276,6 @@ public long pageSizeBytes() {
    *
    * Returns `null` if there was not enough memory to allocate the page. May return a page that
    * contains fewer bytes than requested, so callers should verify the size of returned pages.
-   *
-   * @throws TooLargePageException
```
Comment (srowen) [inline core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java]: Another caveat I didn't mention is that it _is_ a little useful to document _unchecked_ exceptions this way, as only checked exceptions appear in the javadoc. I'd leave this, even without a description.

**Review by srowen — CHANGES_REQUESTED** [review]

File: sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/auth/HiveAuthFactory.java (hunk 1 above)
Comment (turboFei) [reply sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/auth/HiveAuthFactory.java]: ok, thanks.

File: core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java (hunk 2 above)
Comment (turboFei) [reply core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java]: oh, my mistake.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**